src/main/utils/http.ts

```diff
@@ -20,26 +20,40 @@ export async function downloadFile(url: string, file: string, options: RequestOp
     );
 }
 
-export function simpleReq(url: string, options: RequestOptions = {}) {
+export function simpleReq(
+    url: string,
+    options: RequestOptions = {},
+    retries = 0,
+    backoffDelay = 1000
+): Promise<IncomingMessage> {
     return new Promise<IncomingMessage>((resolve, reject) => {
         get(url, options, res => {
             const { statusCode, statusMessage, headers } = res;
             if (statusCode! >= 400) return void reject(`${statusCode}: ${statusMessage} - ${url}`);
             if (statusCode! >= 300) return simpleReq(headers.location!, options).then(resolve).catch(reject);
-
             resolve(res);
+        }).on("error", err => {
+            if (retries > 10) {
+                reject(err);
+            } else {
+                setTimeout(() => resolve(simpleReq(url, options, retries + 1, backoffDelay * 2)), backoffDelay);
+            }
```

Comment:
Shouldn't this be instead recursing backwards until 0?
The `retries` parameter would instead default to 10.

Reply:
Refactored it https://github.com/Vencord/Vesktop/pull/432/commits/eaabc3aae9a8fc44f1887c5dc4c4a2537e190515